imageproxy.go

```diff
@@ -234,7 +238,9 @@ func (p *Proxy) serveImage(w http.ResponseWriter, r *http.Request) {
 	copyHeader(w.Header(), resp.Header, "Cache-Control", "Last-Modified", "Expires", "Etag", "Link")
 
 	if req.Options.TTL != 0 {
-		w.Header().Set("Cache-Control", fmt.Sprintf("max-age=%d", req.Options.TTL))
+		if p.allowedResponseToCache(resp.StatusCode) {
```

Comment:
Shouldn't we do this for error responses https://stackoverflow.com/a/71778683/543280 ?

Also https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/HTTPStatusCodes.html